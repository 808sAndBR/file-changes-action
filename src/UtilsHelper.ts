import {setFailed} from '@actions/core'
import * as t from '@actions/core'
import {ActionError} from './typings/ActionError'

export function getErrorString(
  name: string,
  status = 500,
  from: string,
  message: string,
  error: any = ''
): string {
  try {
    const test = JSON.stringify({
      error: `${status}/${name}`,
      from,
      message,
      payload: error
    } as ActionError, null, 2)
    return test
  } catch (error_) {
    setFailed(`Error throwing error.\n ${JSON.stringify(error_.message)}`)
    throw(new Error(JSON.stringify({name: '500/undefined', message: 'Error throwing error.'})))
  }
}