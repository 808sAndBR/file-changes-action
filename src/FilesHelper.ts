import { writeFileSync } from 'fs'
import { setOutput as coreSetOutput, debug as coreDebug } from '@actions/core'
import type { GithubFile } from './GithubHelper'
import {getErrorString} from './UtilsHelper'

export interface ChangedFiles {
  [key: string]: string[]
}

/**
 * @function sortChangedFiles
 * @param files pass in array of GithubFile's to be sorted
 * @returns ChangedFiles object that has .files, .added, .modified, and .deleted
 */
export function sortChangedFiles(files: GithubFile[]): ChangedFiles {
  try {
    coreDebug(JSON.stringify(files, null, 2))
    const changedFiles: ChangedFiles = files.reduce((acc: ChangedFiles, f: GithubFile) => {
      acc[f.status].push(f.filename || (f.added || f.removed || f.modified))
      acc.files.push(f.filename || (f.added || f.removed || f.modified))
      return acc
    }, {} as ChangedFiles)
    return changedFiles
  } catch (error) {    
    const eString = `There was an issue sorting files changed files.`
    throw new Error(getErrorString(error.name || 'unknown', error.status, sortChangedFiles.name, eString, JSON.stringify(error)))
  }
}

/**
 * @function getFormatExt
 * @param format output format 'json' = '.json' ',' = '.csv' anything else is '.txt'.
 * @returns file extension, '.json', '.csv', or '.txt'
 */
export function getFormatExt(format: string): string {
  let ext
  switch (format.trim()) {
    case 'json':
      ext = '.json'
      break
    case ',':
      ext = '.csv'
      break
    default:
      ext = '.txt'
      break
  }
  return ext
}

/**
 * @function formatChangedFiles
 * @param format output format 'json' will stringify anything else will files.join('string')
 * @param files string list of files to format
 * @returns string for output of changedFiles
 */
export function formatChangedFiles(format: string, files: string[]): string {
  if (format === 'json') {
    return JSON.stringify(files)
  } 
  return files.join(format)
  
}

/**
 * @function writeFiles
 * @param format output format 'json' will stringify anything else will files.join('string')
 * @param key changedFiles type added, modified, deleted, or files
 * @param files string list of files to format
 * @returns string output to be stored in file
 */
export function writeFiles(format: string, key: string, files: string[]): void {
  try {
    const ext = getFormatExt(format)
    const fileName = ((key === 'files') ? `${key}${ext}` : `files_${key}${ext}`)
    coreDebug(`Writing output file ${process.env.HOME}/${fileName}${ext} with ${format} and files ${JSON.stringify(files, null, 2)}`)
    writeFileSync(
      `${process.env.HOME}/${fileName}`,
      formatChangedFiles(format, files),
      'utf-8'
    )
  } catch (error) {
    const eString = `There was an issue writing output files.`
    throw new Error(getErrorString(error.name || 'unknown', error.status, writeFiles.name, eString, JSON.stringify(error)))
  }
}

/**
 * @function writeOutput
 * @param format output format 'json' will stringify anything else will files.join('string')
 * @param key changedFiles type added, modified, deleted, or files
 * @param files string list of files to format
 * @returns string output to be stored to action output
 */
export function writeOutput(format: string, key: string, files: string[]): void {
  try {
    const fileName = ((key === 'files') ? key : `files_${key}`)
    coreDebug(`Writing output ${fileName} with ${format} and files ${JSON.stringify(files, null, 2)}`)
    coreSetOutput(fileName, formatChangedFiles(format, files))
  } catch (error) {
    const eString = `There was an issue setting action outputs.`
    throw new Error(getErrorString(error.name || 'unknown', error.status, writeOutput.name, eString, JSON.stringify(error)))
  }
}
