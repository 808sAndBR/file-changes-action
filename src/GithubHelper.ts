import {GitHub} from '@actions/github'
import type { Inferred } from './InputHelper'
import { getErrorString } from './UtilsHelper'

export interface GithubFile {
  added: string
  modified: string
  removed: string
  filename: string
  status: string
  previous_filename: string
  distinct: boolean
}

/**
 * @function initClient
 * @throws {Error} not sure what might trigger this, but it will throw an error.
 * @param token github token to add to client
 * @returns authenticated github client
 */
export function initClient(
  token: string
): GitHub {
  try {
    return new GitHub(token)
  } catch (error) {
    const eString = `There was an error creating github client.  \
Please check your token.`
    throw new Error(getErrorString(error.name, error.status, initClient.name, eString, error))
  }
}
/**
 * @function getChangedPRFiles
 * @throws {Error} when a 404 or other is received.  404 can be bad repo, owner, pr, or unauthenticated
 * @param client authenticated github client (possibly un-authenticated if public)
 * @param repo  repo string. file-changes-action
 * @param owner owner string.  trilom
 * @param pullNumber pr number to get changed files for
 * @returns Promise of array of changed files
 */
export async function getChangedPRFiles(
  client: GitHub,
  repo: string,
  owner: string,
  pullNumber: number
): Promise<GithubFile[]> {
  try {
    const options = client.pulls.listFiles.endpoint.merge({
      owner, repo, pull_number: pullNumber
    })
    const files: GithubFile[] = await client.paginate(
      options,
      response => response.data
    )
    return files
  } catch(error) {
    const eString = `There was an error getting change files for 
repo:${repo} owner:${owner} pr:${pullNumber}`
    let ePayload: string
    if (error.name === 'HttpError' && +error.status === 404) ePayload = getErrorString(error.name, error.status, getChangedPRFiles.name, eString, error)
    else ePayload = getErrorString(`Unknown Error:${error.name || ''}`, error.status, getChangedPRFiles.name, eString, error.message)
    throw new Error(ePayload)
  }
}
/**
 * @function getChangedPushFiles
 * @throws {Error} when a 404 or other is received.  404 can be bad repo, owner, sha, or unauthenticated
 * @param client authenticated github client (possibly un-authenticated if public)
 * @param repo  repo string. file-changes-action
 * @param owner owner string.  trilom
 * @param base BASE commit sha to compare
 * @param head HEAD commit sha to compare
 * @returns Promise of array of changed files
 */
export async function getChangedPushFiles(
  client: GitHub,
  repo: string,
  owner: string,
  base: string,
  head: string
): Promise<GithubFile[]> {
  try {
    const options = client.repos.compareCommits.endpoint.merge({
      owner, repo, base, head
    })
    const files:GithubFile[] = await client.paginate(
      options,
      response => response.data.files
    )
    return files
  } catch (error) {
    const eString = `There was an error getting change files for 
repo:${repo} owner:${owner} base:${base} head:${head}`
    let ePayload: string  
    if (error.name === 'HttpError' && +error.status === 404) ePayload = getErrorString(error.name, error.status, getChangedPushFiles.name, eString, error)
    else ePayload = getErrorString(`Unknown Error:${error.name || ''}`, error.status, getChangedPushFiles.name, eString, error.message)
    throw new Error(ePayload)
  }
}
/**
 * @function getChangedFiles
 * @param client client authenticated github client (possibly un-authenticated if public)
 * @param repoFull repo owner/repo string.  trilom/file-changes-action
 * @type {Inferred} pass in iinferred type from inferInput
 * @returns Promise of an array of changed PR or push files
 */
export async function getChangedFiles(
  client: GitHub,
  repoFull: string,
  { before, after, pr = NaN }: Inferred
): Promise<GithubFile[]> {
  try {
    if (repoFull.split('/').length > 2) 
      throw new Error(getErrorString(`Bad-Repo`, 500, getChangedFiles.name, `Repo input of ${repoFull} has more than 2 length after splitting.`))
    const owner = repoFull.split('/')[0]
    const repo = repoFull.split('/')[1]
    let files:GithubFile[] = []
    if (Number.isNaN(pr)) 
      files = await getChangedPushFiles(client, repo, owner, before || '', after || '')
    else 
      files = await getChangedPRFiles(client, repo, owner, pr)
    return files
  } catch (error) {
    const pError = JSON.parse(error.message)
    if (pError.from.includes('getChanged')) 
      throw new Error(JSON.stringify({ ...pError, ...{ from: `${error.status}/${error.name}`} }, null, 2))
    const eString = `There was an error getting change files outputs 
pr: ${pr} before: ${before} after: ${after}`
    const ePayload: string = getErrorString(`Unknown Error:${error.name || ''}`, error.status, getChangedFiles.name, eString, error.message)
    throw new Error(ePayload)
  }
}