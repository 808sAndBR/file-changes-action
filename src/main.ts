// External Dependencies
import {error as coreError, setFailed as coreSetFailed} from '@actions/core'
import { writeOutput, writeFiles, sortChangedFiles} from './FilesHelper'
import {getInputs, inferInput} from './InputHelper'
import { getChangedFiles, initClient} from './GithubHelper'


// figure out if it is a PR or Push
export async function run(): Promise<void> {
  try {
    // get inputs
    const inputs = getInputs()
    // parse input
    const inferred = inferInput(inputs.pushBefore, inputs.pushAfter, inputs.prNumber)
    // prepare client
    const client = initClient(inputs.githubToken)
    // get changed files
    const changedFilesArray = await getChangedFiles(client, inputs.githubRepo, inferred)
    // sort changed files
    const changedFiles = sortChangedFiles(changedFilesArray)
    Object.keys(changedFiles).forEach(key => {
      // write file output
      writeFiles(inputs.fileOutput, key, changedFiles[key])
      // write output vars
      writeOutput(inputs.output, key, changedFiles[key])
    })
    // eslint-disable-next-line unicorn/no-process-exit
    process.exit(0)
  } catch (error) {
    const pError = JSON.parse(error.message)
    const prettyError = JSON.stringify(error.message, null, 2)
    // catch error from getInputs
    if (pError.from.includes(getInputs.name))
      coreError(`There was an getting action inputs.\nException: ${prettyError}`)
    // catch error from inferInput
    if (pError.from.includes(inferInput.name))
      coreError(`There was an issue inferring inputs to the action.\nException: ${prettyError}`)
    // catch error from initClient
    else if (pError.from.includes(initClient.name))
      coreError(`There was an issue initilizing the github client.\nException: ${prettyError}`)
    // catch error from getChangedFiles
    else if (pError.from.includes(getChangedFiles.name))
      coreError(`There was an issue getting changed files from Github.\nException: ${prettyError}`)
    // catch error from sortChangedFiles
    else if (pError.from.includes(sortChangedFiles.name))
      coreError(`There was an issue sorting changed files from Github.\nException: ${prettyError}`)
    // catch error from writeFiles
    else if (pError.from.includes(writeFiles.name))
      coreError(`There was an issue writing output files.\nException: ${prettyError}`)
    // catch error from writeOutput
    else if (pError.from.includes(writeOutput.name))
      coreError(`There was an issue writing output variables.\nException: ${prettyError}`)
    else coreError(JSON.stringify(pError))
    coreSetFailed(error.message)
  }
}

run()
