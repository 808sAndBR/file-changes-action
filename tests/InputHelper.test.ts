import { Env, p } from './mocks/env/env'
let env: Env // env object

describe.each(p.testEvents)('Testing %s event...', event => {
  beforeAll(() => env = new Env({}, {githubToken: 'TestToken'}, event))
  afterEach(() => {
    process.env = {...env.envStart}
    jest.resetModules()
    env = new Env({}, {}, event)
  })
  it('Sets correct default input parameters.', () => {
    let {payload, issue} = env.contextMock
    let { prNumber, pushAfter, pushBefore, githubToken, githubRepo, output, fileOutput } = require('../src/InputHelper').getInputs()
    let getInput = require('@actions/core').getInput
    // evaluate outputs
    if (event.includes('push')) {
      expect(prNumber).toBe(NaN)
      expect(pushAfter).toBe(payload.after)
      expect(pushBefore).toBe(payload.before)
    }
    if (event.includes('pull_request') || event.includes('issue_comment')) {
      expect(prNumber).toBe(issue.number)
      if (event === 'pull_request_synchronize') {
        expect(pushAfter).toBe(payload.after)
        expect(pushBefore).toBe(payload.before)
      } else {
        expect(pushAfter).toBeFalsy()
        expect(pushBefore).toBeFalsy()
      }
    }
    expect(githubToken).toBe(process.env.INPUT_GITHUBTOKEN)
    expect(githubRepo).toBe(process.env.GITHUB_REPOSITORY)
    expect(output).toBe('json')
    expect(fileOutput).toBe('json')
    expect(getInput).toHaveBeenCalledTimes(7)
  })
  it('Throws error with no token (undefined) process.env["GITHUB_TOKEN"] \
or (undefined) input githubToken', () => {
    delete process.env.GITHUB_TOKEN
    delete process.env.INPUT_GITHUBTOKEN
    let getInput = require('@actions/core').getInput
    expect(() => {
      require('../src/InputHelper').getInputs()
    }).toThrowErrorMatchingSnapshot()
    expect(getInput).toHaveBeenCalledTimes(1)
  })
  it('Throws error with empty string ("") process.env["GITHUB_TOKEN"] \
or empty string ("") input githubToken', () => {
    env.updateInput({ githubToken: '' })
    process.env.GITHUB_TOKEN = ''
    let getInput = require('@actions/core').getInput
    expect(() => {
      require('../src/InputHelper').getInputs()
    }).toThrowErrorMatchingSnapshot()
    expect(getInput).toHaveBeenCalledTimes(1)
  })
  // test getInputs function
  it.each(p.getTestEvents(p.inputTestInputs, event))
  ('Sets %s input "%s" should be %p', (inputName, input, expected) => {
    env.updateInput({[inputName]: input})
    let {payload, issue} = env.contextMock
    let { prNumber, pushAfter, pushBefore, githubToken, githubRepo, output, fileOutput } = require('../src/InputHelper').getInputs()
    let getInput = require('@actions/core').getInput
    // evaluate outputs
    if (event.includes('push')) {
      expect(prNumber).toBe((inputName === 'prNumber') ? expected : NaN)
      expect(pushAfter).toBe((inputName === 'pushAfter') ? expected : payload.after )
      expect(pushBefore).toBe((inputName === 'pushBefore') ? expected : payload.before)
    }
    if (event.includes('pull_request') || event.includes('issue_comment')) {
      expect(prNumber).toBe((inputName === 'prNumber') ? expected : issue.number)
      if (event === 'pull_request_synchronize') {
        expect(pushAfter).toBe((inputName === 'pushAfter') ? expected : payload.after )
        expect(pushBefore).toBe((inputName === 'pushBefore') ? expected : payload.before)
      } else {
        expect(pushAfter).toBe((inputName === 'pushAfter') ? expected : false )
        expect(pushBefore).toBe((inputName === 'pushBefore') ? expected : false )
      }
    }
    expect(githubToken).toBe((inputName === 'githubToken') ? expected : 'EnvDefaultToken')
    expect(githubRepo).toBe((inputName === 'githubRepo') ? expected : process.env.GITHUB_REPOSITORY)
    expect(output).toBe((inputName === 'output') ? expected : 'json')
    expect(fileOutput).toBe((inputName === 'fileOutput') ? expected : 'json')
    expect(getInput).toHaveBeenCalledTimes(7)
  })
  // test inferInput function
  it.each(p.getTestEvents(p.inferTestInputs, event))
  ('%s', (title, input, expected) => {
    let error = require('@actions/core').error
    let warning = require('@actions/core').warning
    if (title.includes('ERROR')) {
      expect(() => {
        require('../src/InputHelper').inferInput(input.before, input.after, input.pr)  
      }).toThrowErrorMatchingSnapshot()
    } else {
      let data = require('../src/InputHelper').inferInput(input.before, input.after, input.pr)
      Object.keys(data).forEach((key) => expect(data[key]).toBe(expected[key]))
      expect(error).not.toHaveBeenCalled()
    }
    // evaluate outputs
    if (title.includes('WARN weird')) 
      expect(warning).toHaveBeenCalledWith(expect.stringContaining(`received a before(${input.before}) or after(${input.after}) value.`))
    if (title.includes('WARN all')) 
      expect(warning).toHaveBeenCalledWith(expect.stringContaining(`but received a before(${input.before}), after(${input.after}), and PR(${input.pr}).`))
    else 
      expect(error).not.toHaveBeenCalled()
  })
})
