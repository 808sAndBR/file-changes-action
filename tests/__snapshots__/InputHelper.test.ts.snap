// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing issue_comment_created event... Throws error with empty string ("") process.env["GITHUB_TOKEN"] or empty string ("") input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_TOKEN\\": \\"\\",
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/issue_comment_created.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"issue_comment\\",
    \\"INPUT_GITHUBTOKEN\\": \\"\\"
  }
}"
`;

exports[`Testing issue_comment_created event... Throws error with no token (undefined) process.env["GITHUB_TOKEN"] or (undefined) input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/issue_comment_created.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"issue_comment\\"
  }
}"
`;

exports[`Testing issue_comment_created event... throws ERROR with no inputs NOT PUSH or PULL_REQUEST event 1`] = `
"{
  \\"error\\": \\"500/inferInput Error\\",
  \\"from\\": \\"inferInput\\",
  \\"message\\": \\"Received event from issue_comment, but received no inputs. {event_name:issue_comment, pr: NaN, before:, after:}\\",
  \\"payload\\": \\"\\"
}"
`;

exports[`Testing issue_comment_edited event... Throws error with empty string ("") process.env["GITHUB_TOKEN"] or empty string ("") input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_TOKEN\\": \\"\\",
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/issue_comment_edited.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"issue_comment\\",
    \\"INPUT_GITHUBTOKEN\\": \\"\\"
  }
}"
`;

exports[`Testing issue_comment_edited event... Throws error with no token (undefined) process.env["GITHUB_TOKEN"] or (undefined) input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/issue_comment_edited.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"issue_comment\\"
  }
}"
`;

exports[`Testing issue_comment_edited event... throws ERROR with no inputs NOT PUSH or PULL_REQUEST event 1`] = `
"{
  \\"error\\": \\"500/inferInput Error\\",
  \\"from\\": \\"inferInput\\",
  \\"message\\": \\"Received event from issue_comment, but received no inputs. {event_name:issue_comment, pr: NaN, before:, after:}\\",
  \\"payload\\": \\"\\"
}"
`;

exports[`Testing pull_request_opened event... Throws error with empty string ("") process.env["GITHUB_TOKEN"] or empty string ("") input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_TOKEN\\": \\"\\",
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/pull_request_opened.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"pull_request\\",
    \\"INPUT_GITHUBTOKEN\\": \\"\\"
  }
}"
`;

exports[`Testing pull_request_opened event... Throws error with no token (undefined) process.env["GITHUB_TOKEN"] or (undefined) input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/pull_request_opened.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"pull_request\\"
  }
}"
`;

exports[`Testing pull_request_reopened event... Throws error with empty string ("") process.env["GITHUB_TOKEN"] or empty string ("") input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_TOKEN\\": \\"\\",
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/pull_request_reopened.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"pull_request\\",
    \\"INPUT_GITHUBTOKEN\\": \\"\\"
  }
}"
`;

exports[`Testing pull_request_reopened event... Throws error with no token (undefined) process.env["GITHUB_TOKEN"] or (undefined) input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/pull_request_reopened.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"pull_request\\"
  }
}"
`;

exports[`Testing pull_request_synchronize event... Throws error with empty string ("") process.env["GITHUB_TOKEN"] or empty string ("") input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_TOKEN\\": \\"\\",
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/pull_request_synchronize.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"pull_request\\",
    \\"INPUT_GITHUBTOKEN\\": \\"\\"
  }
}"
`;

exports[`Testing pull_request_synchronize event... Throws error with no token (undefined) process.env["GITHUB_TOKEN"] or (undefined) input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/pull_request_synchronize.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"pull_request\\"
  }
}"
`;

exports[`Testing push event... Throws error with empty string ("") process.env["GITHUB_TOKEN"] or empty string ("") input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_TOKEN\\": \\"\\",
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/push.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"push\\",
    \\"INPUT_GITHUBTOKEN\\": \\"\\"
  }
}"
`;

exports[`Testing push event... Throws error with no token (undefined) process.env["GITHUB_TOKEN"] or (undefined) input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/push.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"push\\"
  }
}"
`;

exports[`Testing push_merge event... Throws error with empty string ("") process.env["GITHUB_TOKEN"] or empty string ("") input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_TOKEN\\": \\"\\",
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/push_merge.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"push\\",
    \\"INPUT_GITHUBTOKEN\\": \\"\\"
  }
}"
`;

exports[`Testing push_merge event... Throws error with no token (undefined) process.env["GITHUB_TOKEN"] or (undefined) input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/push_merge.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"push\\"
  }
}"
`;

exports[`Testing schedule event... Throws error with empty string ("") process.env["GITHUB_TOKEN"] or empty string ("") input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_TOKEN\\": \\"\\",
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/schedule.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"schedule\\",
    \\"INPUT_GITHUBTOKEN\\": \\"\\"
  }
}"
`;

exports[`Testing schedule event... Throws error with no token (undefined) process.env["GITHUB_TOKEN"] or (undefined) input githubToken 1`] = `
"{
  \\"error\\": \\"500/getInputs Error\\",
  \\"from\\": \\"getInputs\\",
  \\"message\\": \\"Received an issue getting action inputs.\\",
  \\"payload\\": {
    \\"GITHUB_WORKSPACE\\": \\"/Users/bkillian/repos/file-changes-action/tests/workspace/github\\",
    \\"GITHUB_REPOSITORY\\": \\"trilom-test/file-changes-action\\",
    \\"GITHUB_ACTION\\": \\"file-changes-action\\",
    \\"GITHUB_EVENT_PATH\\": \\"/Users/bkillian/repos/file-changes-action/tests/payloads/events/schedule.json\\",
    \\"GITHUB_EVENT_NAME\\": \\"schedule\\"
  }
}"
`;

exports[`Testing schedule event... throws ERROR with no inputs NOT PUSH or PULL_REQUEST event 1`] = `
"{
  \\"error\\": \\"500/inferInput Error\\",
  \\"from\\": \\"inferInput\\",
  \\"message\\": \\"Received event from schedule, but received no inputs. {event_name:schedule, pr: NaN, before:, after:}\\",
  \\"payload\\": \\"\\"
}"
`;
